tests/integration/run_tests.py

```diff
@@ -67,13 +67,15 @@ def run_single_test(name, test, test_label="", buffer_length=50):
         "stdout": stdout,
         "stderr": stderr,
         "return_code": return_code,
+        "violated_condition": None,
     }
 
     # ensure all test conditions are satisfied
-    for condition in conditions:
+    for i, condition in enumerate(conditions):
         condition.ingest_info(info)
         passed = condition.passes
         if passed is False:
```

Comment:
Yes, I think it will be as easy as that. Then there's the question of either making each test (currently a dictionary entry) into a function, or maybe even finding a way to load the dictionary into dynamic function-maker that pytest can see and run.

Reply:
Yeah, pytest encourages decorators to make functors pretty easily.